jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -975,15 +989,18 @@ public boolean content(ByteBuffer buffer)
 
             _requestBuffer.retain();
             stream._chunk = Content.Chunk.asChunk(buffer, false, _requestBuffer);
+            _delayedForContent = false;
             return true;
         }
 
         @Override
         public boolean contentComplete()
         {
-            // Do nothing at this point.
+            // Do nothing at this point unless we delayed for content
```

Comment:
```suggestion
            // Do nothing at this point unless we delayed for content.
```

Not on this line, but the `_delayedForContent` logic has not been ported to `earlyEOF()` nor for idle timeouts  like it was in Jetty 11.